### PR TITLE
Fms io to fms tests

### LINKED ISSUE #544: FMS2: Need to move some functions into fms_mod from fms_io_mod
**Is your feature request related to a problem? Please describe.**
As discussed in the FMS I/O Meeting on July 20th, the following subroutines are in ```fms_io_mod```.  As we move from the use of ```fms_io_mod``` to ```fms2_io_mod``` in GFDL_atmos_cubed_sphere dycore, these functions will need to be used still and are not in ```fms2_io_mod```.  
~~[set_domain](https://github.com/NOAA-GFDL/FMS/blob/master/fms/fms_io.F90#L7417)
[nullify_domain](https://github.com/NOAA-GFDL/FMS/blob/master/fms/fms_io.F90#L7437)~~
[string](https://github.com/NOAA-GFDL/FMS/blob/master/fms/fms_io.F90#L526)

**Describe the solution you'd like**
The group decided that these should be moved from ```fms_io_mod``` to ```fms_mod```.

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by thomas-robinson**
**Description**
1. Moves functions string_from_integer, string_from_real, ~~set_domain, and nullify_domain~~ to fms.F90
2. Removes references to fms_io_initialized
3. Updates string_from_integer to use a more modern Fortran approach
4. Removes the fms_io tests
5. Adds a test of the string function

Fixes #544 

**How Has This Been Tested?**
`make` with intel 2020 on amd test machine.  `make check` fails, but my new test passes.  (`make check` fails on the master branch also).

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by uramirez8707 — CHANGES_REQUESTED** [review]

File: fms/fms.F90 [hunk 1]
```diff
@@ -998,6 +1013,68 @@ function monotonic_array ( array, direction )
...
+
+  end function string_from_real
+
+  !#######################################################################
+
+
+
+!> \brief set_domain is called to save the domain2d data type prior to
+!! calling the distributed data I/O routines, read_data and write_data.
+subroutine set_domain (Domain2)
```
[the remark this answers is not in the document]
Comment (thomas-robinson) [reply fms/fms.F90]: I updated #544 to have ~~set domain~~ and ~~nullify_domain~~.  I removed them from this PR.

**Review by wrongkindofdoctor — APPROVED** [review]

File: fms/fms.F90 [hunk 2]
```diff
@@ -202,8 +202,8 @@ module fms_mod
           open_file, open_direct_file
 
 ! routines for reading/writing distributed data
-public :: set_domain, read_data, write_data, read_compressed, read_distributed
-public :: get_domain_decomp, field_size, nullify_domain
```
Comment (uramirez8707) [inline fms/fms.F90:206]: You forgot to bring these back. 

Most model components that use this have 
`use          fms_mod, only: set_domain, nullify_domain`

**Review by uramirez8707 — CHANGES_REQUESTED** [review]

File: fms/fms.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (laurenchilutti) [reply fms/fms.F90]: Yes, my testing works when we remove set_domain and nullify_domain.

File: fms/fms.F90 (hunk 2 above)
Comment (thomas-robinson) [reply fms/fms.F90:206]: You're right!  Thanks for catching this.

**Review by uramirez8707 — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]

**Review by laurenchilutti — APPROVED** [review]

**Pull request merged**